libs/design/src/molecules/sidebar/sidebar-viewport/content-shift.ts

```diff
@@ -0,0 +1,29 @@
+import { QueryList } from '@angular/core';
+
+import { DaffSidebarComponent } from '../sidebar/sidebar.component';
+
+/**
+ * Given a list of sidebars, compute the associated content shift.
+ *
+ * Currently, this content shift only occurs if there is an "under" sidebar.
+ * Importantly, since there can be "under" sidebars on either side, we compute
+ * a negative translation in the opposite direction if the sidebar is on the right.
+ */
+export const sidebarViewportContentShift = (sidebars: QueryList<DaffSidebarComponent>): number =>
+  sidebars.reduce((acc: number, sidebar) => {
+    if(sidebar.mode !== 'under' || !sidebar.open) {
+      return acc;
+    }
+
+    if(sidebar.side === 'left'){
+      return sidebar.width;
+    } else if (sidebar.side === 'right' && acc === 0) {
+      return -1 * sidebar.width;
```

Comment:
Should you be adding these values to `acc`? I'm not sure why else this would be implemented with `reduce`. Otherwise it will just use the contribution of the last sidebar and throw away the previous values (although maybe that is intended?).

Reply:
Yeah, this is unclear, I meant to let the last `left` in a given chain be the highest priority.